consensus/consortium/common/contract.go

```diff
@@ -494,13 +494,13 @@ func ApplyTransaction(msg types.Message, opts *ApplyTransactOpts) (err error) {
 		// The system transaction is the transaction that have the receiver is ConsortiumV2Contracts
 		actualTx := (*receivedTxs)[0]
 		if !bytes.Equal(signer.Hash(actualTx).Bytes(), expectedHash.Bytes()) {
-			return fmt.Errorf("expected tx hash %v, get %v, nonce %d, to %s, value %s, gas %d, gasPrice %s, data %s", expectedHash.String(), actualTx.Hash().String(),
-				expectedTx.Nonce(),
-				expectedTx.To().String(),
-				expectedTx.Value().String(),
-				expectedTx.Gas(),
-				expectedTx.GasPrice().String(),
-				hex.EncodeToString(expectedTx.Data()),
+			return fmt.Errorf(
+				"expected transaction: nonce %d, to %s, value %s, gas %d, gasPrice %s, data %s\n"+
```

Comment:
should we also log tx hash here

Reply:
The reason tx's hash is not included into the log is that transaction hash is a result of hash on the whole transaction which includes the signature. When mismatch happens, the local node finds out that the local computed tx is different from remote tx, but it can't compute the signature value of its local computed tx. So it cannot correctly compute the local computed tx hash.

Reply:
added